crates/flux-infer/src/fixpoint_encoding.rs

```diff
@@ -1851,3 +1944,57 @@ fn mk_implies(assumption: fixpoint::Pred, cstr: fixpoint::Constraint) -> fixpoin
         Box::new(cstr),
     )
 }
+
+struct SexpParseCtxt {
+    kvar_args: FxIndexSet<String>,
+}
+
+impl FromSexp<FixpointTypes> for SexpParseCtxt {
+    fn kvar(&self, _name: &str) -> Result<fixpoint::KVid, ParseError> {
+        todo!()
+    }
+
+    fn string(&self, _s: &str) -> Result<fixpoint::SymStr, ParseError> {
+        todo!()
+    }
+
+    fn var(&self, name: &str) -> Result<fixpoint::Var, ParseError> {
+        match self.kvar_args.get_index_of(name) {
```

Comment:
If we ever want to parse non-cut vars (or counterexamples), we'll need to extend the parse context here to include all of the variables we synthesize or convert during the fixpoint encoding.

Reply:
Not just non-cut vars, qualifiers can contain (global) constants.

```rust
const N: usize = 1;

defs! {
    qualifier MyQual(x: int) { x > N }
}
```